extensions/validation/src/main/java/org/mule/extension/validation/internal/CommonValidationOperations.java

```diff
@@ -0,0 +1,519 @@
+     * @param options         the {@link ValidationOptions}
+     * @param event           the current {@link MuleEvent
+     */
+    @Operation
+    public void url(String url,
+                    @Optional(defaultValue = "true") boolean allowTwoSlashes,
+                    @Optional(defaultValue = "true") boolean allowAllSchemes,
+                    @Optional(defaultValue = "true") boolean allowLocalUrls,
+                    @Optional(defaultValue = "false") boolean noFragments,
+                    @ParameterGroup ValidationOptions options, MuleEvent event) throws Exception
+    {
+        ValidationContext context = createContext(options, event);
+        validateWith(new UrlValidator(url, allowTwoSlashes, allowAllSchemes, allowLocalUrls, noFragments, context), context, event);
+    }
+
+    /**
+     * Validates that {@code value} matches the {@code regex} regular expression
+     *
+     * @param value         the value to check
+     * @param regex         the regular expression to check against
+     * @param caseSensitive when {@code true} matching is case sensitive, otherwise matching is case in-sensitive
+     * @param options       the {@link ValidationOptions}
+     * @param event         the current {@link MuleEvent
+     */
+    @Operation
+    public void matchesRegex(String value,
+                             String regex,
+                             @Optional(defaultValue = "true") boolean caseSensitive,
+                             @ParameterGroup ValidationOptions options, MuleEvent event) throws Exception
+    {
+        ValidationContext context = createContext(options, event);
+        validateWith(new MatchesRegexValidator(value, regex, caseSensitive, context), context, event);
+    }
+
+    /**
+     * Perform a list of nested validation operations and informs only one {@link ValidationResult}
+     * which summarizes all of the found errors (if any).
+     * <p/>
+     * If {@code throwsException} is {@code true}, then the {@link ValidationResult}
+     * is communicated by throwing a {@link ValidationException}. On the other hand,
+     * if {@code throwsException} is {@code false}, then the {@link ValidationResult}
+     * is set as the message payload.
+     * <p/>
+     * When configured through XML, all the {@code validations} must include
+     * the All the child processors must contain the {@code validator-message-processor}
+     * substitution group.
+     *
+     * @param validations     the nested validation operations
+     * @param throwsException if {@code true}, then a {@link ValidationException} is thrown if any validator fails.
+     * @param muleEvent       the current {@link MuleEvent}
+     * @return the same {@code muleEvent} that was passed as argument
+     * @throws ValidationException if at least one validator fails and {@code throwsException} is {@code true}
+     */
+    @Operation
+    public MuleEvent all(@RestrictedTo(ValidationExtension.class) List<NestedProcessor> validations,
```

Comment:
This method should be moved to a different class. It's not a type of validation but a validation execution strategy.